### DISCUSSION TITLE
State space equations for SPMe

### DISCUSSION THREAD

**Discussion by Mrzhang-hub** [pybamm-team/PyBaMM#2771] [Q&A] [answered]
I want to apply SPMe to solve some control problems, e.g. fast charging, which needs the state space equations. However, I could not find related information in this package. I wonder if there exists these equations and if not, how to accomplish these equations through pybamm? Look forward to your reply! @tinosulzer  @brosaplanella

**Comment by brosaplanella** [comment, marked as answer]
The state space equations can be derived from the Jacobian of the system (see [Jacobian linearization in these slides](https://courses.engr.illinois.edu/ece486/fa2021/documentation/lectures/slides/Lecture25B_Linearization.pdf)). You could extract the Jacobian from a discretised PyBaMM model (see [the docs](https://pybamm.readthedocs.io/en/latest/source/api/expression_tree/operations/jacobian.html) for more info).

**Reply by Mrzhang-hub** [reply]
Thanks for your fast response and I will dive into the detailed derivation process of state space equations of SPMe.

**Reply by brosaplanella** [reply]
It would be good to have a notebook of this as an example, so that should be a nice way to contribute if you are interested.

**Reply by Mrzhang-hub** [reply]
Sure, I am pleasant to write them into notebook and share it into the community after I fully understand the whole process.
